### ISSUE TITLE
Creating a cluster with Amazon Linux 2 doesn't work because of certs path

### ISSUE THREAD

**Issue by cjbottaro** [kubernetes/kops#5673]
**1. What `kops` version are you running?**
```
$ kops version
Version 1.10.0
```

**2. What Kubernetes version are you running?**
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.2", GitCommit:"bb9ffb1654d4a729bb4cec18ff088eacc153c239", GitTreeState:"clean", BuildDate:"2018-08-08T16:31:10Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.1", GitCommit:"b1b29978270dc22fecc592ac55d903350454310a", GitTreeState:"clean", BuildDate:"2018-07-17T18:43:26Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
```

**3. What cloud provider are you using?**

AWS

**4. What commands did you run?  What is the simplest way to reproduce this issue?**
```
kops create -f kops/$CLUSTER_NAME/cluster.yaml
kops create secret --name $CLUSTER_NAME sshpublickey admin -i ~/.ssh/$CLUSTER_NAME.pub
kops update cluster $CLUSTER_NAME --yes
```

Where the instancegroup manifests have:
```
image: amazon/amzn2-ami-hvm-2.0.20180622.1-x86_64-gp2
```

**5. What happened after the commands executed?**

The `k8s-ec2-srcdst` pod has errors.

**9. Anything else do we need to know?**

It all boils down to the manifest for `k8s-ec2-srcdst` specifies the wrong `hostPath` for the certs in Amazon Linux 2. It should be this:
```
volumes:
  - hostPath:
    path: /etc/ssl/certs/ca-bundle.crt # or /etc/pki/tls/certs/ca-bundle.crt ?
    type: ""
    name: ssl-certs
```

I think this is related to #4718 and https://github.com/kubernetes/kops/pull/5577 (see the linked issue and PR in that PR).

**Comment by cjbottaro** [comment]
I don't think that path is correct in my previous message. Even after updating the `k8s-ec2-srcdst` deployment with the previously mentioned path, my cluster is still borked:

```
$ kops validate cluster
Using cluster from kubectl context: edge.aworks.us

Validating cluster edge.aworks.us

INSTANCE GROUPS
NAME			ROLE	MACHINETYPE	MIN	MAX	SUBNETS
infra			Node	m5.large	2	10	us-east-1c
master-us-east-1b	Master	t2.medium	1	1	us-east-1b
master-us-east-1c	Master	t2.medium	1	1	us-east-1c
master-us-east-1d	Master	t2.medium	1	1	us-east-1d

NODE STATUS
NAME				ROLE	READY
ip-10-3-4-4.ec2.internal	master	True
ip-10-3-5-130.ec2.internal	master	True
ip-10-3-5-161.ec2.internal	node	True
ip-10-3-5-225.ec2.internal	node	True
ip-10-3-6-118.ec2.internal	master	True

VALIDATION ERRORS
KIND	NAME					MESSAGE
Pod	kube-system/kube-dns-6b4f4b544c-j6dns	kube-system pod "kube-dns-6b4f4b544c-j6dns" is not healthy
Pod	kube-system/kube-dns-6b4f4b544c-j6dns	kube-system pod "kube-dns-6b4f4b544c-j6dns" is not healthy
Pod	kube-system/kube-dns-6b4f4b544c-j6dns	kube-system pod "kube-dns-6b4f4b544c-j6dns" is not healthy
Pod	kube-system/kube-dns-6b4f4b544c-wcs2p	kube-system pod "kube-dns-6b4f4b544c-wcs2p" is not healthy

Validation Failed

pwd: ~
cjbottaro@kaby$ kubectl get -n kube-system pods
NAME                                                 READY     STATUS             RESTARTS   AGE
aws-node-848f5                                       1/1       Running            0          52m
aws-node-fxv4n                                       1/1       Running            0          50m
aws-node-t48sc                                       1/1       Running            0          50m
aws-node-tb8bq                                       1/1       Running            0          52m
aws-node-tc5g4                                       1/1       Running            0          52m
dns-controller-6844d67498-phpnk                      1/1       Running            0          52m
etcd-server-events-ip-10-3-4-4.ec2.internal          1/1       Running            0          52m
etcd-server-events-ip-10-3-5-130.ec2.internal        1/1       Running            0          51m
etcd-server-events-ip-10-3-6-118.ec2.internal        1/1       Running            0          51m
etcd-server-ip-10-3-4-4.ec2.internal                 1/1       Running            1          51m
etcd-server-ip-10-3-5-130.ec2.internal               1/1       Running            0          51m
etcd-server-ip-10-3-6-118.ec2.internal               1/1       Running            0          52m
k8s-ec2-srcdst-99776fb46-nrgj8                       1/1       Running            0          49m
kube-apiserver-ip-10-3-4-4.ec2.internal              1/1       Running            1          52m
kube-apiserver-ip-10-3-5-130.ec2.internal            1/1       Running            3          51m
kube-apiserver-ip-10-3-6-118.ec2.internal            1/1       Running            2          52m
kube-controller-manager-ip-10-3-4-4.ec2.internal     1/1       Running            0          51m
kube-controller-manager-ip-10-3-5-130.ec2.internal   1/1       Running            0          51m
kube-controller-manager-ip-10-3-6-118.ec2.internal   1/1       Running            0          51m
kube-dns-6b4f4b544c-j6dns                            0/3       CrashLoopBackOff   26         52m
kube-dns-6b4f4b544c-wcs2p                            2/3       CrashLoopBackOff   28         50m
kube-dns-autoscaler-6b658bd4d5-cwvqx                 1/1       Running            0          52m
kube-proxy-ip-10-3-4-4.ec2.internal                  1/1       Running            0          52m
kube-proxy-ip-10-3-5-130.ec2.internal                1/1       Running            0          51m
kube-proxy-ip-10-3-5-161.ec2.internal                1/1       Running            0          49m
kube-proxy-ip-10-3-5-225.ec2.internal                1/1       Running            0          50m
kube-proxy-ip-10-3-6-118.ec2.internal                1/1       Running            0          51m
kube-scheduler-ip-10-3-4-4.ec2.internal              1/1       Running            0          51m
kube-scheduler-ip-10-3-5-130.ec2.internal            1/1       Running            0          52m
kube-scheduler-ip-10-3-6-118.ec2.internal            1/1       Running            0          52m
```

Or maybe that's a whole other issue? The logs for the `kube-dns` pods just say:
```
I0712 09:37:11.953661       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:12.453626       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
E0712 09:37:12.454204       1 reflector.go:199] k8s.io/dns/vendor/k8s.io/client-go/tools/cache/reflector.go:94: Failed to list *v1.Endpoints: Get https://10.3.0.1:443/api/v1/endpoints?resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E0712 09:37:12.454254       1 reflector.go:199] k8s.io/dns/vendor/k8s.io/client-go/tools/cache/reflector.go:94: Failed to list *v1.Service: Get https://10.3.0.1:443/api/v1/services?resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
I0712 09:37:12.953657       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:13.453653       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
```

Over and over.

**Comment by mattiasek1** [comment]
Any solution here? I have the same issues.

**Comment by cjbottaro** [comment]
It's all issues with `amazon-vpc-ini`, and I don't even think it's specific to Amazon Linux 2.

Using `kube-router` for networking with an Amazon Linux 2 cluster works completely fine.

Using `amazon-vpc-ini` with any of the following images results an unstable cluster:
```
kope.io/k8s-1.11-debian-stretch-amd64-hvm-ebs-2018-05-27
kope.io/k8s-1.9-debian-stretch-amd64-hvm-ebs-2018-03-11
amazon/amzn2-ami-hvm-2.0.20180622.1-x86_64-gp2
```

And by unstable, I mean the DNS pods will _eventually_ stop being able to connect to the API server:
```
I0712 09:37:11.953661       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:12.453626       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
E0712 09:37:12.454204       1 reflector.go:199] k8s.io/dns/vendor/k8s.io/client-go/tools/cache/reflector.go:94: Failed to list *v1.Endpoints: Get https://10.3.0.1:443/api/v1/endpoints?resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E0712 09:37:12.454254       1 reflector.go:199] k8s.io/dns/vendor/k8s.io/client-go/tools/cache/reflector.go:94: Failed to list *v1.Service: Get https://10.3.0.1:443/api/v1/services?resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
I0712 09:37:12.953657       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:13.453653       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
```

Other strangeness I've noticed with `amazon-vpc-ini`... you don't want to bring up nodes in the cluster until `api.internal.your.cluster` starts to resolve, especially if you want to run your DNS pods on non-master nodes.

After `api.internal.your.cluster` resolves, then bring up worker nodes, start DNS pods on them, and they work _for a while_. That's the part I don't get. After a bit of time (maybe 20-30 mins), they stop being able to talk to the api server.

tl;dr Solution is to not use `amazon-vpc-ini`. Amazon Linux 2 works fine with other networking (at least `kube-router`).

**Comment by dannyvargas23** [comment]
I have had the same issue using the amazoncni. It only works flawlessly using the debian-jessie images. I experienced many DNS issues using the debian-stretch image.

**Comment by saorio** [comment]
I am also spending time on this issue. 
I reffered #4331 mistaken certificate file path and amazon cni image changed `0.1.4`.
But kube-dns is still CrashLoopBackOff.
